industry_benchmarks/utils/results_cleanup.py

```diff
@@ -0,0 +1,349 @@
+            elif protocol_unit_result_count == 0:
+                print("All protocol units failed, skipping")
+                continue
+
+            # get the name of the key which is a gufe token
+            # for the only ProtocolUnitResult-* in unit_results
+            # this means we can grab the first that matches since there is only
+            # one ProtocolUnitResult-*
+            proto_key = next(
+                k
+                for k in results["unit_results"].keys()
+                if k.startswith("ProtocolUnitResult")
+            )
+
+            results_dir = (
+                Path(results["unit_results"][proto_key]["outputs"]["nc"]["path"])
+                .resolve()
+                .parent
+            )
+            # if the dir doesn't exist, we should try and fix it
+            if not results_dir.is_dir():
+                print("Fixing path to results dir")
+                results_dir = remove_first_reversed_sequential_duplicate_from_path(
+                    results_dir
+                )
+
+                # Now we should check if the dir exists
+                if not results_dir.is_dir():
+                    print("Can't find results directory, skipping")
+                    continue
+
+            structural_analysis_data = results["unit_results"][proto_key]["outputs"][
+                "structural_analysis"
+            ]
+
+            # save structural analysis data
+            print("Saving structural analysis data")
+            np.savez_compressed(
+                results_dir / "structural_analysis_data.npz",
+                protein_RMSD=np.asarray(
+                    structural_analysis_data["protein_RMSD"], dtype=np.float32
+                ),
+                ligand_RMSD=np.asarray(
+                    structural_analysis_data["ligand_RMSD"], dtype=np.float32
+                ),
+                ligand_wander=np.asarray(
+                    structural_analysis_data["ligand_wander"], dtype=np.float32
+                ),
+                protein_2D_RMSD=np.asarray(
+                    structural_analysis_data["protein_2D_RMSD"], dtype=np.float32
+                ),
+                time_ps=np.asarray(
+                    structural_analysis_data["time(ps)"], dtype=np.int32
+                ),
+            )
+
+            # remove structural_analysis data stuffed into unit results
+            del results["unit_results"][proto_key]["outputs"]["structural_analysis"]
+
+            # Now we sub sample the traj and save reporter data
+            simulation = results_dir / "simulation.nc"
+            checkpoint = results_dir / "checkpoint.chk"
+            hybrid_pdb = results_dir / "hybrid_system.pdb"
+            # TODO better name?
+            outfile = results_dir / "energy_replica_state.npz"
+            out_traj = results_dir / "out"
+            print("Subsampling trajectory and saving energy data")
+            extract_data(simulation, checkpoint, hybrid_pdb, outfile, out_traj)
```

Comment:
Before we advertise we should check on some vacuum sims that this is all good for the energy data.

Reply:
good point, I think I only checked this on solvent and complex sims

Reply:
Maybe to explain my thoughts here - I was thinking we should subsample, then re-read the data and estimate the dG from the new files. If it works then we know that we haven't forgotten anything.

Reply:
OH! Sure, can you send me a snippet that does it and I can wire it in? I can even put that in the `info.yaml` so it will be really easy to slurp up the results (wouldn't need gather)